modules/integration/tests-integration/tests-amqp/src/test/java/org/wso2/mb/integration/tests/amqp/functional/TransactionalPublishingTestCase.java

```diff
@@ -0,0 +1,489 @@
+        // Creating clients
+        consumerClient1 = new AndesClient(consumerConfig, true);
+        consumerClient1.startClient();
+
+        publisherClient1 = new AndesClient(publisherConfig, true);
+        AndesJMSPublisher publisher = publisherClient1.getPublishers().get(0);
+
+        Message message = publisher.getSession().createTextMessage("transactional message");
+        publisher.getSender().send(message);
+
+        TimeUnit.MILLISECONDS.sleep(AndesClientConstants.DEFAULT_RUN_TIME);
+
+        Assert.assertEquals(consumerClient1.getReceivedMessageCount(), expectedBeforeCommit,
+                "Message received! send messages are not committed hence no message should be received");
+
+        publisher.getSession().commit();
+
+        AndesClientUtils.waitForMessagesAndShutdown(consumerClient1, AndesClientConstants.DEFAULT_RUN_TIME);
+
+        Assert.assertEquals(consumerClient1.getReceivedMessageCount(), expectedCount,
+                "Expected message count not received after commit");
+    }
+
+    /**
+     * Test publisher transaction rollback functionality
+     *
+     * - Create a transactional publisher and a non transactional subscriber
+     * - Enqueue a message
+     * - Test whether subscriber receives a message (Must not receive a message)
+     * - Rollback
+     * - Enqueue a different message
+     * - Commit
+     * - Test whether the subscriber received the message (Must receive)
+     *
+     * @throws AndesClientConfigurationException
+     * @throws IOException
+     * @throws JMSException
+     * @throws AndesClientException
+     * @throws NamingException
+     * @throws InterruptedException
+     */
+    @Test(groups = {"wso2.mb", "queue", "transaction" }, description = "Test for rollback functionality")
+    public void enqueueAndRollbackEnqueueAndCommitTestCase() throws AndesClientConfigurationException,
+            IOException, JMSException, AndesClientException, NamingException, InterruptedException {
+
+        int expectedCount = 1;
+        int expectedBeforeCommit = 0;
+        String transactionMessage = "transactional message";
+        String queueName = "Transactional-enqueueAndRollbackEnqueueAndCommitTestCase";
+
+        // Creating a initial JMS consumer client configuration
+        AndesJMSConsumerClientConfiguration consumerConfig =
+                new AndesJMSConsumerClientConfiguration(ExchangeType.QUEUE, queueName);
+        consumerConfig
+                .setFilePathToWriteReceivedMessages(AndesClientConstants.FILE_PATH_TO_WRITE_RECEIVED_MESSAGES); // writing received messages.
+
+        // Creating a publisher client configuration
+        AndesJMSPublisherClientConfiguration publisherConfig =
+                new AndesJMSPublisherClientConfiguration(ExchangeType.QUEUE, queueName);
+        publisherConfig.setTransactionalSession(true);
+
+        // Creating clients
+        consumerClient1 = new AndesClient(consumerConfig, true);
```

Comment:
No. This is a single publisher test case. That's why methodCleanup has a null check